### PR TITLE
Use the AWS SDK for JavaScript V3

### LINKED ISSUE #1840: Upgrade aws-sdk version
We get the following message by runnin MWoffliner
```
(node:13) NOTE: We are formalizing our plans to enter AWS SDK for JavaScript (v2) into maintenance mode in 2023.

Please migrate your code to use AWS SDK for JavaScript (v3).
For more information, check the migration guide at https://a.co/7PzMCcy
(Use `node --trace-warnings ...` to show where the warning was created)
```

### PR DESCRIPTION
**Pull request by manavg26**
migrated the code to the AWS SDK for JavaScript V3

Tag:  manavg261840-aws-sdk-JSv3

Fixes #1840

### CODE REVIEW

**Comment by kelson42** [comment]
@manavg26 Thank you for your PR, look to the detailed log of the CI to understand why it fails. AFAIK there is at least linting problem. Look the command and start the linting locally and push when this is fixed.

File: package.json [hunk 1]
```diff
@@ -92,6 +93,7 @@
     "imagemin-optipng": "^8.0.0",
     "imagemin-pngquant": "^9.0.2",
     "imagemin-webp": "^7.0.0",
+    "jpegoptim-bin": "^7.1.0",
```
Comment (kelson42) [inline package.json:96]: seems totally unrelated/unnecessary?!

File: src/S3.ts [hunk 2]
```diff
@@ -16,43 +16,45 @@ class S3 {
   }
 
   public async initialise() {
-    const s3UrlBase: any = new S3File.Endpoint(this.url)
-    this.s3Handler = new S3File.S3({
-      endpoint: s3UrlBase,
-      accessKeyId: this.params.keyId,
-      secretAccessKey: this.params.secretAccessKey,
-      s3ForcePathStyle: s3UrlBase.protocol === 'http:',
+    const s3UrlBase: any = new URL(this.url)
```
Comment (kelson42) [inline src/S3.ts:19]: `s3UrlBase` seems not the best name `s3UrlObj` seems better.

File: src/S3.ts [hunk 3]
```diff
@@ -16,43 +16,45 @@ class S3 {
...
                             Bucket used: ${this.bucketName}
                             End point used: ${s3UrlBase.href}
-                            Public IP used: ${await publicIpv4()}`)
-    }
+                            Public IP used: ${publicIpv4()}`)
+      })
   }
 
-  public async bucketExists(bucket: string): Promise<any> {
+  public bucketExists(bucket: string): Promise<any> {
```
Comment (kelson42) [inline src/S3.ts:39]: You have removed the `async` here, what is the rational behind?

**Review by kelson42 — CHANGES_REQUESTED** [review]
First pass of review

**Comment by kelson42** [comment]
@manavg26 For me the feature is broken, if I launch MWoffliner with S3 configuration I get:
```
**********

Unable to connect to S3, either S3 login credentials are wrong or bucket cannot be found
                            Bucket used: xxxxx
                            End point used: xxxx
                            Public IP used: [object Promise]

**********
```

File: src/S3.ts (hunk 3 above)
Comment (manavg26) [reply src/S3.ts:39]: no particular reason seemed a little redundant here.

File: src/S3.ts (hunk 2 above)
Comment (manavg26) [reply src/S3.ts:19]: will change

File: src/S3.ts (hunk 3 above)
Comment (kelson42) [reply src/S3.ts]: Not sure myself, but this code has to apply asynchrnously, so redundant with which part of the code exactl?

**Comment by kelson42** [comment]
@manavg26 what about my last comment that it does not work for me (without th pr it works)? Do i need to change some5ing in my configuration?

**Comment by manavg26** [comment]
> @manavg26 what about my last comment that it does not work for me (without th pr it works)? Do i need to change some5ing in my configuration?

Have made some changes, can you check now?

**Review by kelson42 — CHANGES_REQUESTED** [review]
Still failing

**Comment by kelson42** [comment]
Transfering ownership of issue/PR revieing to @VadimKovalenkoSNF. At some point this has to be fixed/reviewed so we can avoid warning and soon outage because of S3 backend.

File: src/S3.ts [hunk 4]
```diff
@@ -16,43 +17,51 @@ class S3 {
...
 
   public async initialise() {
-    const s3UrlBase: any = new S3File.Endpoint(this.url)
-    this.s3Handler = new S3File.S3({
-      endpoint: s3UrlBase,
-      accessKeyId: this.params.keyId,
-      secretAccessKey: this.params.secretAccessKey,
-      s3ForcePathStyle: s3UrlBase.protocol === 'http:',
+    const s3UrlObj: any = new URL(this.url)
+    const regionRegex: RegExp = /^https?:\/\/s3\.([^.]+)\.amazonaws\.com/;
```
Comment (VadimKovalenkoSNF) [inline src/S3.ts:21]: `amazonaws` is not a valid sub-domain for test purposes, there should be `wasabisys` instead. Also, It should be configurable and not present in the code.

File: src/S3.ts [hunk 5]
```diff
@@ -16,43 +17,51 @@ class S3 {
...
+      },
+      endpoint: s3UrlObj.href,
+      forcePathStyle: s3UrlObj.protocol === 'http:',
     })
-    try {
-      if ((await this.bucketExists(this.bucketName)) === true) {
-        return true
-      }
-    } catch (err) {
-      throw new Error(`Unable to connect to S3, either S3 login credentials are wrong or bucket cannot be found
```
Comment (VadimKovalenkoSNF) [inline src/S3.ts:31]: This error is not explicit enough, probably **err.message** should be used instead.

File: src/S3.ts (hunk 4 above)
Comment (kelson42) [reply src/S3.ts:21]: OK, please fix PR.

**Comment by kelson42** [comment]
superseeded by #1865
